### DISCUSSION TITLE
Buggy Time-series Visualisation

### DISCUSSION THREAD

**Discussion by jmcapra** [apache/superset#23061] [General]
<img width="1270" alt="image" src="https://user-images.githubusercontent.com/35404590/218304382-72451369-8781-4cd7-a261-b02e9d9bd408.png">

When producing time-series charts with regions of null/missing data, like the one above, Superset displays the strange behaviour of "stretching" out adjacent data points. In the example above, there is no data* between 7:15 and 8:55 - yet this displays as a continuous line between the two points. This is poor visualisation practice as it gives the impression of data existing where it does not. 

Is there any way of configuring a Superset chart so that null values are displayed as 0?

* this is aggregated count data - so for each 5 minute time grain period where the count is 0, I should expect to see 0 reflected on the chart.

**Comment by sfirke** [comment]
You could try Advanced Analytics -> Resample and tell it to use zero for nulls.  I have had mixed results getting that to work but I think it's where this *should* be possible.

**Reply by inctl** [reply]
This only works between data points.
However, if you display a day with a grain and resample by the hour, using only a single data point,
the bar chart will display a large bar with no points at 0.

**Reply by rusackas** [reply]
Incorrect display of data is kind of a **big** deal!

Can you file a GH Issue with a made-up dataset (via SQL data generation, or from example datasets) that illustrates the problem with a particular chart configuration? 

If so, then we can write failing tests and/or a fix around it.
